src/Surfnet/StepupRa/RaBundle/Controller/RaaController.php

```diff
@@ -19,39 +19,47 @@
 namespace Surfnet\StepupRa\RaBundle\Controller;
 
 use Surfnet\StepupMiddlewareClientBundle\Identity\Dto\Identity;
+use Surfnet\StepupRa\RaBundle\Service\ProfileService;
 use Surfnet\StepupRa\RaBundle\Command\SelectInstitutionCommand;
 use Surfnet\StepupRa\RaBundle\Form\Type\SelectInstitutionType;
 use Surfnet\StepupRa\RaBundle\Service\InstitutionConfigurationOptionsService;
 use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
 
 class RaaController extends Controller
 {
     public function institutionConfigurationAction(Request $request)
     {
         $this->denyAccessUnlessGranted(['ROLE_RAA', 'ROLE_SRAA']);
-        $token = $this->get('security.token_storage')->getToken();
 
         $logger = $this->get('logger');
         /** @var Identity $identity */
-        $identity = $token->getUser();
-
-        // Load the RA institutions for the identity that is logged in
-        $options = $this
-            ->getRaListingService()
-            ->createChoiceListFor($identity->id, $token->getIdentityInstitution());
-        $institution = reset($options);
-
-        // SRAA's are usually not RAA for other institutions as they already are SRAA. Show the institution config
-        // of the SRAAs SHO, and let her use the SRAA switcher in order to see config for different institutions.
-        if (empty($options) && $this->isGranted('ROLE_SRAA')) {
-            $institution = $token->getIdentityInstitution();
+        $identity = $this->getUser();
+
+        $profile = $this->getProfileService()->findByIdentityId($identity->id);
+
+        $choices = [];
+        foreach ($profile->authorizations as $institution => $role) {
+            if ($role[0] == 'raa') {
```

Comment:
Is this a requirement? That only the RAA locations are displayed in the choices?

Reply:
In order to see configuration's you have to be administrator.  If you are RA you are even not allowed to see the page.

Reply:
That I understand, I was concerned that a RAA, who is RA at another institution is now unable to view the institution configuration of that institution (where he is RA)